tests/kernel/queue/src/test_queue_contexts.c

```diff
@@ -430,3 +432,97 @@ void test_access_kernel_obj_with_priv_data(void)
 					NULL, NULL, NULL, 0, K_USER, K_NO_WAIT);
 	k_thread_join(&tdata, K_FOREVER);
 }
+
+static void low_prio_wait_for_queue(void *p1, void *p2, void *p3)
+{
+	struct k_queue *q = p1;
+	uint32_t *ret = NULL;
+
+	ret = k_queue_get(q, K_FOREVER);
+	zassert_true(*ret == 0xccc,
+	"The low priority thread get the queue data failed lastly");
+}
+
+static void high_prio_t1_wait_for_queue(void *p1, void *p2, void *p3)
+{
+	struct k_queue *q = p1;
+	uint32_t *ret = NULL;
+
+	ret = k_queue_get(q, K_FOREVER);
+	zassert_true(*ret == 0xaaa,
+	"The highest priority and waited longest get the queue data failed firstly");
+}
+
+static void high_prio_t2_wait_for_queue(void *p1, void *p2, void *p3)
+{
+	struct k_queue *q = p1;
+	uint32_t *ret = NULL;
+
+	ret = k_queue_get(q, K_FOREVER);
+	zassert_true(*ret == 0xbbb,
+	"The higher priority and waited longer get the queue data failed secondly");
+}
+
+/**
+ * @brief Test the action that multithread try to get data from queue
```

Comment:
It would better to change it like below.

"Test multi-threads to get data from a queue."

Reply:
Updated, thanks.